L1Trigger/L1THGCal/plugins/geometries/HGCalTriggerGeometryV9Imp3.cc

```diff
@@ -0,0 +1,938 @@
+  }
+
+  return modules;
+}
+
+HGCalTriggerGeometryV9Imp3::geom_set HGCalTriggerGeometryV9Imp3::getLpgbtsFromModule(const unsigned module_id) const {
+  geom_set lpgbt_ids;
+
+  auto module_itrs = module_to_lpgbts_.equal_range(module_id);
+  for (auto module_itr = module_itrs.first; module_itr != module_itrs.second; module_itr++) {
+    lpgbt_ids.emplace(module_itr->second);
+  }
+
+  return lpgbt_ids;
+}
+
+GlobalPoint HGCalTriggerGeometryV9Imp3::getTriggerCellPosition(const unsigned trigger_cell_det_id) const {
+  unsigned det = DetId(trigger_cell_det_id).det();
+  // Position: barycenter of the trigger cell.
+  Basic3DVector<float> triggerCellVector(0., 0., 0.);
+  const auto cell_ids = getCellsFromTriggerCell(trigger_cell_det_id);
+  // Scintillator
+  if (det == DetId::HGCalHSc) {
+    for (const auto& cell : cell_ids) {
+      triggerCellVector += hscGeometry()->getPosition(cell).basicVector();
+    }
+  }
+  // HFNose
+  else if (det == DetId::HGCalTrigger and
+           HGCalTriggerDetId(trigger_cell_det_id).subdet() == HGCalTriggerSubdetector::HFNoseTrigger) {
+    for (const auto& cell : cell_ids) {
+      HFNoseDetId cellDetId(cell);
+      triggerCellVector += noseGeometry()->getPosition(cellDetId).basicVector();
+    }
+  }
+  // Silicon
+  else {
+    for (const auto& cell : cell_ids) {
+      HGCSiliconDetId cellDetId(cell);
+      triggerCellVector += (cellDetId.det() == DetId::HGCalEE ? eeGeometry()->getPosition(cellDetId)
+                                                              : hsiGeometry()->getPosition(cellDetId))
+                               .basicVector();
+    }
+  }
+  return GlobalPoint(triggerCellVector / cell_ids.size());
+}
+
+GlobalPoint HGCalTriggerGeometryV9Imp3::getModulePosition(const unsigned module_det_id) const {
+  unsigned det = DetId(module_det_id).det();
+  // Position: barycenter of the module.
+  Basic3DVector<float> moduleVector(0., 0., 0.);
+  const auto cell_ids = getCellsFromModule(module_det_id);
+  // Scintillator
+  if (det == DetId::HGCalHSc) {
+    for (const auto& cell : cell_ids) {
+      moduleVector += hscGeometry()->getPosition(cell).basicVector();
+    }
+  }
+  // HFNose
+  else if (det == DetId::Forward && DetId(module_det_id).subdetId() == ForwardSubdetector::HFNose) {
+    for (const auto& cell : cell_ids) {
+      HFNoseDetId cellDetId(cell);
+      moduleVector += noseGeometry()->getPosition(cellDetId).basicVector();
+    }
+  }  // Silicon
+  else {
+    for (const auto& cell : cell_ids) {
+      HGCSiliconDetId cellDetId(cell);
+      moduleVector += (cellDetId.det() == DetId::HGCalEE ? eeGeometry()->getPosition(cellDetId)
+                                                         : hsiGeometry()->getPosition(cellDetId))
+                          .basicVector();
+    }
+  }
+
+  return GlobalPoint(moduleVector / cell_ids.size());
+}
+
+void HGCalTriggerGeometryV9Imp3::fillMaps() {
+  // read json mapping file
+  json mapping_config_;
```

Comment:
Trailing underscore to be removed as it is not a class member

Reply:
Removed